petab/v2/problem.py

```diff
@@ -1024,7 +1023,17 @@ class ProblemConfig(BaseModel):
         description="The base path to resolve relative paths.",
         exclude=True,
     )
-    format_version: VersionNumber = "2.0.0"
+    format_version: str = "2.0.0"
```

Comment:
`str | int` in v1 -- intentional difference?

Reply:
That's how it is in the current schema draft. Although not strictly enforced so far, I think it would make sense to always require major.minor. But that's probably a separate discussion.

Reply:
Since there is no benefit to keeping `1` as an int, could use `str` instead of `str | int`, to use pydantic to handle the type casting.

Requiring a full version string is fine for me, but then we need to provide schemas for each minor version. Makes sense to support minor updates to the format, so I'm also in favor of expecting a full `X.X` or `X.X.X` string. major.minor` as suggested is probably enough since we probably don't need to distinguish "minor" and "patch" updates

Reply:
Agreed, but I'll keep that for a future update.

Reply:
Added the `format_version_tuple` property for a normalized version number number.